src/react/hooks/useRouteData/useRouteData.ts

```diff
@@ -0,0 +1,14 @@
+import { value } from "../../../can"
+import routeData from "../../../canjs/routing/route-data"
+import { useCanObservable } from "../useCanObservable"
+
+export const useRouteData = <Value, SetValueArgument = Value>(keyPath: string) => {
```

Comment:
Second generic argument is optional and defaults to the original value. You can set it to be something else though. This would be required for something like `src/react/Filters/hooks/useStatusFilters/useSelectedStatuses.ts`

The value is a string, but the setter expects an array:

```ts
const [statusesToShow, setStatusesToShow] = useRouteData<string, string[]>("statusesToShow");
```

Reply:
Makes sense, i like it